BuildScript.UnitTests/MetadataValueDictionaryTest.cs

```diff
@@ -25,38 +25,28 @@ namespace Remotion.BuildScript.UnitTests
   public class MetadataValueDictionaryTest
   {
     [Test]
-    public void GetValue_Exists_ValueNotNull ()
+    public void GetKeyValuePair_Exists_ValueNotNull ()
     {
       var rawCollection = new Dictionary<string, string> { { "A", "B" }, { "C", "D" } };
       var collection = new MetadataValueDictionary (rawCollection);
 
-      var value = collection.GetValue (new[] { "A", "E" }, () => new Exception());
+      var kvp = collection.GetKeyValuePair (new[] { "A", "E" }, () => new Exception());
```

Comment:
Am I missing the case-insensitivity check for the GetKeyValuePair tests? Also, should test with upper case and lower case in all APIs to 'show' that it works both ways. you could write silently failing code very easily by doing a tolower when creating the dictionary and then always only writing tests that use lower case strings for the lookup. in the end, we end up with a system that fails when you use uppercase strings when requesting a value. I know, it's paranoid, but it's also a very real concern when it comes to maintainability and guarding code against 'great' ideas :)